generator/src/main/java/com/linkedin/pegasus/generator/TemplateSpecGenerator.java

```diff
@@ -418,47 +425,46 @@ else if (schema instanceof PrimitiveDataSchema)
    */
   private CustomClasses getCustomClasses(DataSchema schema)
   {
+    return getCustomClasses(schema, _customTypeLanguage);
+  }
+
+  public static CustomClasses getCustomClasses(DataSchema schema, String customTypeLanguage)
+  {
     CustomClasses customClasses = null;
     final Map<String, Object> properties = schema.getProperties();
-    final Object java = properties.get(JAVA_PROPERTY);
-    if (java != null)
-    {
-      if (java.getClass() != DataMap.class)
-      {
-        throw new IllegalArgumentException(schema + " has \"java\" property that is not a DataMap");
-      }
-      final DataMap map = (DataMap) java;
-      final Object custom = map.get(CLASS_PROPERTY);
-      if (custom != null)
-      {
-        if (custom.getClass() != String.class)
-        {
-          throw new IllegalArgumentException(schema + " has \"java\" property with \"class\" that is not a string");
-        }
-        // a custom class specification has been found
-        customClasses = new CustomClasses();
-        customClasses.customClass = new ClassTemplateSpec();
-        customClasses.customClass.setFullName((String) custom);
-        if (!allowCustomClass(schema))
-        {
-          throw new IllegalArgumentException(schema + " cannot have custom class binding");
+    if (customTypeLanguage != null) {
+      final Object java = properties.get(customTypeLanguage);
+      if (java != null) {
+        if (java.getClass() != DataMap.class) {
+          throw new IllegalArgumentException(schema + " has \"" + customTypeLanguage + "\" property that is not a DataMap");
         }
-      }
-      // check for coercer class
-      final Object coercerClass = map.get(COERCER_CLASS_PROPERTY);
-      if (coercerClass != null)
-      {
-        if (coercerClass.getClass() != String.class)
-        {
-          throw new IllegalArgumentException(schema + " has \"java\" property with \"coercerClass\" that is not a string");
+        final DataMap map = (DataMap) java;
+        final Object custom = map.get(CLASS_PROPERTY);
+        if (custom != null) {
+          if (custom.getClass() != String.class) {
+            throw new IllegalArgumentException(schema + " has \"" + customTypeLanguage + "\" property with \"class\" that is not a string");
+          }
+          // a custom class specification has been found
+          customClasses = new CustomClasses();
```

Comment:
This all makes sense. I'm okay with `getCustomClasses(DataSchema schema, String customTypeLanguage)` being a temporary work around that get's removed once we have a method that returns all meta information.